### PR TITLE
Implementing PrunePathValues function

### PR DESCRIPTION
**Pull request by tomikazi**

### CODE REVIEW

**Review by adibrastegarnia — APPROVED** [review]

**Review by teone — APPROVED** [review]

**Comment by tomikazi** [comment]
retest this please

File: pkg/controller/proposal/controller.go [hunk 1]
```diff
@@ -480,6 +480,8 @@ func (r *Reconciler) reconcileCommit(ctx context.Context, proposal *configapi.Pr
 			for path, changeValue := range changeValues {
 				config.Values[path] = changeValue
 			}
+			config.Values = tree.PrunePathMap(config.Values, true)
```
Comment (adibrastegarnia) [inline pkg/controller/proposal/controller.go:483]: I think you can operate on change values first before putting it in the config values?

Comment (tomikazi) [reply pkg/controller/proposal/controller.go:483]: I'm not sure I can. The change values could contain just the delta no? We want the prune the entire config as a whole.

Comment (adibrastegarnia) [reply pkg/controller/proposal/controller.go:483]: that is right,

**Review by kuujo — APPROVED** [review]

**Pull request merged**